### ISSUE TITLE
Parse COPYUID (new UID of the message)

### ISSUE THREAD

**Issue by arnisoph** [emersion/go-imap-move#12]
It could be quite useful for the consumer to get the UID of a message, if the message got a different UID (after UID move/UID copy). That doesn't happen with Dovecot, but with nothing more unknown than Gmail. And it took my hours to find it out....

```
0t5PIw SELECT INBOX
* FLAGS (\Answered \Flagged \Draft \Deleted \Seen $MailFlagBit0 $NotPhishing $Phishing 123 asdasd fooooooo forty-two)
* OK [PERMANENTFLAGS (\Answered \Flagged \Draft \Deleted \Seen $MailFlagBit0 $NotPhishing $Phishing 123 asdasd fooooooo forty-two \*)] Flags permitted.
* OK [UIDVALIDITY 1] UIDs valid.
* 1 EXISTS
* 0 RECENT
* OK [UIDNEXT 986] Predicted next UID.
* OK [HIGHESTMODSEQ 63520]
0t5PIw OK [READ-WRITE] INBOX selected. (Success)
lVh-tA UID MOVE 985 "X-POSISTO-Flagging"
* 1 EXPUNGE
* 0 EXISTS
lVh-tA OK [COPYUID 315 985 2] (Success)
```
2 is the new UID of the message.

This is not a bug report but a feature request.

**Comment by emersion** [comment]
Yeah, I've been thinking about this too.

The COPY part has been implemented in https://github.com/emersion/go-imap-uidplus. The interaction between the MOVE and UIDPLUS extensions makes things awkward. The solution will probably be to merge everything into go-imap, see https://github.com/emersion/go-imap/issues/322.

Note, it's still not clear how the API should look like. There's no guarantee the server will support UIDPLUS, and clients should be prepared for this.

There's also no perfect way to get the UID when the extension isn't supported:

> If the server does not return the APPENDUID or COPYUID response
   codes, the client can discover this information by selecting the
   destination mailbox.  The location of messages placed in the
   destination mailbox by COPY or APPEND can be determined by using
   FETCH and/or SEARCH commands (e.g., for Message-ID or some unique
   marker placed in the message in an APPEND).

**Comment by arnisoph** [comment]
Yes, totally broken stuff this damn protocol.... I guess I will work with Message-ID and let the app silently fail if the message id isn't set. I'm not in the mood to workaround (all) MUA/protocol problems. :)
